pkg/analyzer/analyzers/dockerhub/requests.go

```diff
@@ -0,0 +1,115 @@
+package dockerhub
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// LoginResponse is the successful response from the /login API
+type LoginResponse struct {
+	Token string `json:"token"`
+}
+
+// ErrorLoginResponse is the error response from the /login API
+type ErrorLoginResponse struct {
+	Detail        string `json:"detail"`
+	Login2FAToken string `json:"login_2fa_token"` // if login require 2FA authentication
+}
+
+// RepositoriesResponse is the /repositories/<namespace> response
+type RepositoriesResponse struct {
+	Result []struct {
+		Name      string `json:"name"`
+		Type      string `json:"repository_type"`
+		IsPrivate bool   `json:"is_private"`
+		StarCount int    `json:"star_count"`
+		PullCount int    `json:"pull_count"`
+	} `json:"results"`
+}
+
+// login call the /login api with username and jwt token and if successful retrieve the token string and return
+func login(client *http.Client, username, pat string) (string, error) {
+	payload := strings.NewReader(fmt.Sprintf(`{"username": "%s", "password": "%s"}`, username, pat))
+
+	req, err := http.NewRequest(http.MethodPost, "https://hub.docker.com/v2/users/login", payload)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var token LoginResponse
+		if err := json.NewDecoder(resp.Body).Decode(&token); err != nil {
+			return "", err
+		}
+
+		return token.Token, nil
+	case http.StatusUnauthorized:
+		var errorLogin ErrorLoginResponse
+		if err := json.NewDecoder(resp.Body).Decode(&errorLogin); err != nil {
+			return "", err
+		}
+
+		if errorLogin.Login2FAToken != "" {
+			// TODO: handle it more appropriately
```

Comment:
question: Just confirming this TODO is valid, and not necessary for this initial implementation?

Reply:
Yup, it is working. If 2FA is enabled for account we return error. Added TODO just in case we figure out a more better approach in future to handle these scenarios.